common/pipeline.go

```diff
@@ -47,3 +48,26 @@ func (cplMgr *codePipelineManager) ListState(pipelineName string) ([]*codepipeli
 
 	return output.StageStates, nil
 }
+
+func getRevisionFromCodePipeline(pipelineName string) (string, error) {
+	sess := session.Must(session.NewSession())
+	service := codepipeline.New(sess)
```

Comment:
Don't start a new session...reuse connection in `codepipelineiface.CodePipelineAPI`.  This one is already honoring the region and profile that the user requested.

Also, this will improve your ability to unit test.  Check out the unit test for `ListState` and notice how you are able to mock the AWS SDK to avoid having to actually talk to the CodePipeline service.